csharp/src/sdk/Geometry/Polygon.cs

```diff
@@ -268,36 +283,41 @@ public IEnumerable<Polygon> Intersection(Polygon polygon)
         }
 
         /// <summary>
-        /// Returns Polygons representing the attempted Union between this Polygon and the supplied Polygon.
+        /// Constructs the geometric union between this Polygon and the supplied Polygon.
         /// </summary>
         /// <param name="polygon">The Polygon to be combined with this Polygon.</param>
-        /// <returns></returns>
-        public IEnumerable<Polygon> Union(Polygon polygon)
+        /// <returns>
+        /// Returns a single Polygon from a successful union.
+        /// Returns null if a union cannot be performed on the two Polygons.
+        /// </returns>
+        public Polygon Union(Polygon polygon)
         {
-            var thisPath = PolygonExtensions.ToClipperPath(this);
-            var polyPath = PolygonExtensions.ToClipperPath(polygon);
+            var thisPath = this.ToClipperPath();
+            var polyPath = polygon.ToClipperPath();
             Clipper clipper = new Clipper();
             clipper.AddPath(thisPath, PolyType.ptSubject, true);
             clipper.AddPath(polyPath, PolyType.ptClip, true);
             var solution = new List<List<IntPoint>>();
             clipper.Execute(ClipType.ctUnion, solution);
-            var polygons = new List<Polygon>();
-            foreach (List<IntPoint> path in solution)
+            if (solution.Count > 1)
```

Comment:
Is this really true? Does clipper returns a list with the inputs if the union fails?

Reply:
Yeah, it surprised me too. I sent it two disjoint Polygons through my unit test and got 2 Polygons back.

Reply:
There was a clue in the Clipper docs that made me think I'd better test it.